plugins/module_utils/azure_rm_common.py

```diff
@@ -1584,9 +1584,27 @@ def _get_profile(self, profile="default"):
 
         return None
 
-    def _get_msi_credentials(self, subscription_id=None, client_id=None, **kwargs):
-        credentials = MSIAuthentication(client_id=client_id)
-        credential = MSIAuthenticationWrapper(client_id=client_id)
+    def _get_msi_credentials(self, subscription_id=None, client_id=None, _cloud_environment=None, **kwargs):
+        if not _cloud_environment:
+            cloud_environment = azure_cloud.AZURE_PUBLIC_CLOUD
+        else:
+            # try to look up "well-known" values via the name attribute on azure_cloud members
+            all_clouds = [x[1] for x in inspect.getmembers(azure_cloud) if isinstance(x[1], azure_cloud.Cloud)]
+            matched_clouds = [x for x in all_clouds if x.name == _cloud_environment]
+            if len(matched_clouds) == 1:
+                cloud_environment = matched_clouds[0]
+            elif len(matched_clouds) > 1:
+                self.fail("Azure SDK failure: more than one cloud matched for cloud_environment name '{0}'".format(cloud_environment))
+            else:
+                if not urlparse.urlparse(cloud_environment).scheme:
```

Comment:
@doanminhtu  “cloud_environment” Without definition before use, can you refine your changes?
```suggestion
                if not urlparse.urlparse(cloud_environment).scheme:
```

Reply:
hi @Fred-sun . Many thanks to your review and suggestion. My code was not correct. Then I have just fixed the issue and new code pushed. 
Could you please take a look to review it ? 
Best regards,